### DISCUSSION TITLE
Can I use this to test my own service worker implementation?

### DISCUSSION THREAD

**Discussion by dethstrobe** [mswjs/msw#683] [Q&A] [answered]
It sounds like this package is meant to mock service requests from my web app.

But I'm looking to create my own service worker, which, as I have found out is not the easiest thing to unit test and even some integration level testing and e2e testing becomes problematic.

Would msw be a good fit to unit test my own service worker?

I know you can only have 1 service worker registered, so I assume this might even make things more complicated for me.

**Comment by kettanaito** [comment, marked as answer]
Hey, @dethstrobe. 

I don't think you should use MSW to test a custom Service Worker. MSW _uses a Service Worker_ to intercept HTTP requests and then give you control over their responses. The worker is treated as an implementation detail. The library doesn't expose any means to _test a worker_. 

I think you may be looking for something like [service-worker-mock](https://www.npmjs.com/package/service-worker-mock) (pardon the naming confusion). I haven't used that package, but it seems to be designed for testing workers, which is what you're after.
